crates/dojo-core/src/model.cairo

```diff
@@ -1,11 +1,27 @@
 use dojo::world::IWorldDispatcher;
 use starknet::SyscallResult;
 
+/// Trait that is implemented at Cairo level for each struct that is a model.
+trait ModelValues<T> {
+    fn values(self: @T) -> Span<felt252>;
+    fn from_values(values: Span<felt252>) -> T;
+    fn get(world: IWorldDispatcher, id: felt252) -> T;
+    fn set(self: @T, world: IWorldDispatcher, id: felt252);
```

Comment:
```suggestion
    fn update(self: @T, world: IWorldDispatcher, id: felt252);
```
The logic will be Torii indexing a `StoreSetRecord` when `set` is used as we're doing currently.
For this new value system, it's only an `update`, as we already know the keys somehow, hence the model has already been set. 👍